content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md

```diff
@@ -0,0 +1,175 @@
+and used for a long time. Commonly used tool to enable this was created on [Aug 21, 2018](https://github.com/grpc-ecosystem/grpc-health-probe/commit/2df4478982e95c9a57d5fe3f555667f4365c025d) with
+the first release at [Sep 19, 2018](https://github.com/grpc-ecosystem/grpc-health-probe/releases/tag/v0.1.0-alpha.1).
+
+This approach for gRPC apps health checking is very popular. There are [3,626 Dockerfiles](https://github.com/search?l=Dockerfile&q=grpc_health_probe&type=code)
+with the `grpc_health_probe` and [6,621 yaml](https://github.com/search?l=YAML&q=grpc_health_probe&type=Code) files that are discovered with the
+basic search on GitHub (at the moment of writing). This is good indication of the tool popularity
+and the need to support this natively.
+
+With the Kubernetes 1.23, native support for gRPC health probing was introduced as alpha feature.
+The feature is not at beta and enabled by default.
+
+## Using the feature
+
+The gRPC health checking is [easy to use](/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-grpc-liveness-probe).
+The natively supported health probe has many benefits over the workaround involving `grpc_health_probe` executable.
+
+With the native gRPC support you don't need to download and carry `10MB` of an additional executable with your image.
+Exec probes are generally slower than a gRPC call as they require instantiating a new process to run an executable.
+It also makes the checks less sensible for edge cases when the pod is running at maximum resources and has troubles
+instantiating new processes.
+
+There are a few limitations though. Since configuring a client certificate for probes is hard,
+services that require client authentication are not supported. The built-in probes are also
+not checking the server certificates and ignore related problems.
+
+Built-in checks also cannot be configured to ignore certain types of errors
+(`grpc_health_probe` returns different exit codes for different errors),
+and cannot be "chained" to run the health check on multiple services in a single probe.
+
+But all these limitations are quite standard for gRPC and there are easy workarounds
+for those.
+
+## Try it for yourself
+
+### Getting cluster
+
+You can try this feature today. To try the feature you can spin up the kubernetes cluster
+yourself with the `GRPCContainerProbe` feature gate enabled.
+
+Alternatively, at the moment of writing, you can spin up the cluster on GKE
+using the following command (note, version is `1.23` and `enable-kubernetes-alpha` are specified):
+
+``` bsh
+$ gcloud container clusters create test-grpc \
+    --enable-kubernetes-alpha \
+    --no-enable-autorepair \
+    --no-enable-autoupgrade \
+    --release-channel=rapid \
+    --cluster-version=1.23
+```
+
+You will also need to configure `kubectl` to access the cluster:
+
+``` bsh
+$ gcloud container clusters get-credentials test-grpc
+```
```

Comment:
- we should explain how to get a v1.24 cluster, not a v1.23 cluster
- we shouldn't require or imply that the user uses Google Cloud